pkg/spanconfig/spanconfigtestutils/utils.go

```diff
@@ -236,6 +240,19 @@ func PrintSpanConfigDiffedAgainstDefaults(conf roachpb.SpanConfig) string {
 	if !reflect.DeepEqual(conf.LeasePreferences, defaultConf.LeasePreferences) {
 		diffs = append(diffs, fmt.Sprintf("lease_preferences=%v", conf.VoterConstraints))
 	}
+	if !reflect.DeepEqual(conf.ProtectedTimestamps, defaultConf.ProtectedTimestamps) {
+		sort.Slice(conf.ProtectedTimestamps, func(i, j int) bool {
```

Comment:
Can we avoid sorting? I think this mutates the underlying array, and I'm not sure we need it for test determinism.